### PR TITLE
Remove the use of AbstracLifecycleComponent constructor Backport#37488

### PR DESCRIPTION
**Pull request by pgomulka**
The AbstracLifecycleComponent used to extend AbstractComponent, so it had to pass settings to the constractor of its supper class.
It no longer extends the AbstractComponent so there is no need for this constructor
There is also no need for AbstracLifecycleComponent subclasses to have Settings in their constructors if they were only passing it over to super constructor.
This is part 1. which will be backported to 6.x with a migration guide/deprecation log.
part 2 will have this constructor removed in 7
relates #35560

relates #34488

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
run the gradle build tests 1

**Comment by pgomulka** [comment]
run the gradle build tests 1

File: server/src/main/java/org/elasticsearch/common/component/AbstractLifecycleComponent.java [hunk 1]
```diff
@@ -34,6 +34,13 @@
 
     private final List<LifecycleListener> listeners = new CopyOnWriteArrayList<>();
 
+    protected AbstractLifecycleComponent() {}
+
+    /**
+     * @deprecated the settings parameters are not used, therefore the use of this constructor is deprecated.
```
Comment (danielmitterdorfer) [inline server/src/main/java/org/elasticsearch/common/component/AbstractLifecycleComponent.java]: Can you please indicate which constructor should be called instead (for completeness)?

**Review by danielmitterdorfer — APPROVED** [review]
LGTM. I left a minor comment around the Javadoc that explains the deprecation.

**Comment by pgomulka** [comment]
run the gradle build tests 1

**Comment by pgomulka** [comment]
run the gradle build tests 1

**Comment by pgomulka** [comment]
run the gradle build tests 1

**Pull request merged**